diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8324372..b48da69 100755
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -18,8 +18,8 @@ option(BUILD_SHARED_LIBS "Build using shared libraries" ON)
 set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED True)
 
-set(ADDITIONAL_SHARED_LIBRARIES_TO_LINK
-  libstdc++fs.so)
+#set(ADDITIONAL_SHARED_LIBRARIES_TO_LINK
+#  libstdc++fs.so)
 
 set(HIPCC_BIN
   hipcc)
